ubuntu_image/classic_builder.py

```diff
@@ -77,23 +80,26 @@ def prepare_image(self):
             super().prepare_image()
 
     def populate_rootfs_contents(self):
-        src = os.path.join(self.unpackdir, 'chroot')
         dst = self.rootfs
-        for subdir in os.listdir(src):
-            shutil.move(os.path.join(src, subdir), os.path.join(dst, subdir))
-        # Remove default grub bootloader settings as we ship bootloader bits
-        # (binary blobs and grub.cfg) to a generated rootfs locally.
-        grub_folder = os.path.join(dst, 'boot', 'grub')
```

Comment:
Why has this been removed? It's a rather big architectural change for just adding a feature.
This is certainly something that would have to be discussed in more detail before we can proceed.

Reply:
What happened with this is that doing `sudo update-grub` fails if there is no /boot/grub folder. It is not terribly important, because updating grub deb packages works anyway, but why is the reason for removing the folder? I do not see the impact of that.

Reply:
We want to make sure that the bootloader bits installed to the image are what is specified in the gadget snap, not just what the .deb in the image might happen to install by default.  This doesn't necessarily require us to remove the /boot/grub *directory*, but I think it is correct to remove all the *contents* of that directory.

But I expect the gadget snap processing to recreate /boot/grub before the image is finalized.  Is update-grub being called from somewhere before this happens?

Reply:
Ok, I would say it like this: I think we need to re-visit this part of the code for sure, but this is not the place for it. I'm against including multiple unrelated feature-changes in one PR, it's hard to later on grasp what is what and revert if needed. Could you strip out the grub related changes to a separate PR? We would then merge this instantly.